File: plugins/plugin-elizacloud/python/elizaos_plugin_elizacloud/models/speech.py
from elizaos_plugin_elizacloud.providers import ElizaCloudClient
from elizaos_plugin_elizacloud.types import ElizaCloudConfig, TextToSpeechParams


async def handle_text_to_speech(
    config: ElizaCloudConfig,
    params: TextToSpeechParams,
) -> bytes:
    async with ElizaCloudClient(config) as client:
        return await client.generate_speech(params)

File: plugins/plugin-elizacloud/rust/src/models/speech.rs
#![allow(missing_docs)]

use crate::error::Result;
use crate::providers::ElizaCloudClient;
use crate::types::{ElizaCloudConfig, TextToSpeechParams};

pub async fn handle_text_to_speech(
    config: ElizaCloudConfig,
    params: TextToSpeechParams,
) -> Result<Vec<u8>> {
    let client = ElizaCloudClient::new(config)?;
    client.generate_speech(params).await
}
